### PR TITLE
Add Window centering and monitor size.

### PR DESCRIPTION
**Pull request by mralve**
## Description

Made so windows will now be centered on the current monitor by default.
Small fixes in ScreenDimensions docs to clarify.

## Additions

DisplayConfig - Added `centered_window` bool to be able to toggle the centering.
ScreenDimensions - Added monitor HEIGHT and WIDTH functions to get the current screens res.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Ran `cargo test --all` locally if this modified any rs files.
- [x] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [x] Updated the content of the book if this PR would make the book outdated.
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [x] Added unit tests for new APIs if any were added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: amethyst_renderer/src/config.rs [hunk 1]
```diff
@@ -44,6 +45,9 @@ pub struct DisplayConfig {
     /// Maximum window dimensions, measured in pixels (px).
     pub max_dimensions: Option<(u32, u32)>,
 
+    /// Whether the window will be centered on the monitor by deafult.
```
Comment (jojolepro) [inline amethyst_renderer/src/config.rs]: Typo in default

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -36,6 +36,8 @@ it is attached to. ([#1282])
 * Adding support for AMETHYST_NUM_THREADS environment variable to control size of the threads pool used by thread_pool_builder.
 * Add `Input` variant to `StateEvent`. ([#1478])
 * Support type parameters in `EventReader` derive. ([#1478])
+* Added `monitor_width` and `monitor_height` in `ScreenDimensions`. ([#1539])
+* Added `centered_window` bool to `DisplayConfig` Windows will now be centered by deafult. ([#1539])
```
Comment (jojolepro) [inline docs/CHANGELOG.md]: Please add the link to the pull request. See the lines under those ones to see how.

**Comment by azriel91** [comment]
Wondering if this should be in before or after 0.11 ? (breaking change)
Also, am not an expert but should the monitor dimensions be tracked in a separate type (i.e. what if you move the window to a different monitor at runtime, then the stored values aren't consistent)

**Comment by mralve** [comment]
@azriel91 sorry about that, it now uses the get_current_monitor() and is updated by the window_management() so the values should track the change of the screen if that occurs.

**Comment by mralve** [comment]
@azriel91 about 0.11 I am unsure, it would in the worst case be a minor fix to enter centered_window bool in display config? but that's on by default.

**Comment by azriel91** [comment]
ah, re:0.11, it's probably better answered by @Frizi / @fhaynes (rendy integration is nigh, this might creep the scope); otherwise i'd like it in 🙃.

p/s; could you make the 1280 and 800 public constants in `amethyst_test`? I use the screen constants in my extended version of the test crate

**Comment by fhaynes** [comment]
@azriel91 as long as you handle integrating it with rendy, go ahead.

**Comment by Frizi** [comment]
The whole renderer is going to be replaced, including window management being split into it's own crate.
I'm perfectly OK with the change itself, but i would prefer to merge it once rendy branch is merged into master. 

Other option is to make a PR against `rendy` branch so we can merge it immediately :)

**Comment by mralve** [comment]
I guess we can dismiss this PR then, and move this to Rendy then @Frizi Expect a new PR soon :)

**Comment by fhaynes** [comment]
Closing this PR per @mralve and @Frizi doing it on the rendy branch. =) Thanks!

**Pull request closed without merging**